### PR TITLE
Fix os_image for mac build in semaphore.yml

### PR DESCRIPTION
**Pull request by Lord-Kamina**
### What does this PR do?

Updates the os_image definition in semaphore.yml.

### More

- [ ] Added/updated documentation

### CODE REVIEW

**Comment by Lord-Kamina** [comment]
Oh; fuck; they're deprecating the macOS 10.14 image; and here I've been doing everything in my power to avoid upgrading to Catalina.

**Comment by Lord-Kamina** [comment]
And, it's fixed; I aalso took the liberty of updating the GCC and Clang versions we use for Linux builds.

File: .semaphore/semaphore.yml
Comment (OznOg) [inline .semaphore/semaphore.yml:61]: why is macport on a dropbox instead of some "official" repository?

Comment (Lord-Kamina) [reply .semaphore/semaphore.yml:61]: It's my Dropbox hahaha, it's because it's the already built ports; to save time essentially.

**Comment by OznOg** [comment]
LGTM if it does the job

File: .semaphore/semaphore.yml
Comment (Baklap4) [inline .semaphore/semaphore.yml:15]: For what would we need this ppa?

File: .semaphore/semaphore.yml
Comment (Baklap4) [reply .semaphore/semaphore.yml:61]: Hmm maybe create a repo + release for this containing that tarball? This way it's listed on github, saving you space on dropbox and for everyone to see?

File: .semaphore/semaphore.yml
Comment (Baklap4) [reply .semaphore/semaphore.yml:15]: As mentioned on discord we need it for Boost.ASIO which is in ubuntu 18.04 at 1.65 while we at least require boost 1.66

Ubuntu 19.04 is shipped with boost 1.67

Comment (Baklap4) [reply .semaphore/semaphore.yml:15]: @performous/core-owners What do we think of using a personal more updated ppa?
I'm not quite fond of it myself but if it's necessaryit's necessary..
I'm not fond of it for the reason as it might mess up the users system since we're not OS maintainers we can't really tell the impact. And aside from this we also need to keep our documentation up to date regarding what and how to install stuff properly so you can build performous

Comment (Lord-Kamina) [reply .semaphore/semaphore.yml:15]: As I said in Discord; my issue is that despite being arguably the most popular Linux flavor, Ubuntu keeps some really outdated packages here and there and sometimes the reasons are dubious at best... for example, take LibXML++; there is no sane reason for the current cutting-edge of ubuntu still having only LibXML++2.6; and yet it does (seeing how, even if some packages would break by using 3.0; they could ship both, since they're designed to coexist).

With regards to boost, Back when 18.04 first came out, 1.65 was already almost a full year old, 1.66 was out and 1.67 came out around the same time as Ubuntu 18.04. I just don't see it as justifiable that in TWO-YEARS of their "LTS", they didn't update the goddamn boost version that was already obsolete when it first came out.

I think it's ridiculous to hold updating our code when even Windows has easily-accessible libraries that are kept more updated.

I might have missed something of course but from a bit of search into package-dependencies, the only thing that might be broken by adding the updated icu and boost packages (in addition to that ppa, they are also available on separate ppas to minimize new packages) to bionic is an old version of OpenTTD (another game)

Do keep in mind that packages in launchpad have to build before they are published, and if they won't build if they break anything important in the first place.

**Review by nieknooijens — APPROVED** [review]

**Pull request merged**
